modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDeletedHandler.es.js

```diff
@@ -133,12 +142,18 @@ export const handleFieldDeleted = (props, state, {activePage, fieldName}) => {
 
 	const newPages = pages.map((page, pageIndex) => {
 		if (activePage === pageIndex) {
+			const newPages = removeField(
+				props,
+				pages,
+				fieldName,
+				removeEmptyRows
+			);
+
 			return {
 				...page,
-				rows: FormSupport.removeEmptyRows(
-					removeField(props, pages, fieldName),
-					pageIndex
-				),
+				rows: removeEmptyRows
+					? FormSupport.removeEmptyRows(newPages, pageIndex)
+					: newPages[pageIndex].rows,
```

Comment:
@carolmariaabb, do we need this ternary operator? If we just invoke `FormSupport.removeEmptyRows(newPages, pageIndex)` isn't enough once it's already returning the proper `rows` according to `removeEmptyRows` parameter?

Reply:
@natocesarrego The problem with calling the `removeEmptyRows` function is that if you add a new field and resize it to the left, and then try to move this field to the right column, as described at [LPS-114356](https://issues.liferay.com/browse/LPS-114356), since you removed it from the row, which now has two empty columns, the `removeEmptyRows` will delete this empty row before you can move to the right column.

Reply:
I see, @carolmariaabb. Let's keep it. 🙂